lab_2_keywords_cooccurrence/main.py

```diff
@@ -3,12 +3,39 @@
 Extract keywords based on co-occurrence frequency
 """
 from pathlib import Path
-from typing import Optional, Sequence, Mapping
+from string import punctuation
+from typing import Optional, Sequence, Mapping, Any
 
 KeyPhrase = tuple[str, ...]
 KeyPhrases = Sequence[KeyPhrase]
 
 
+def check_list(user_input: Any, element_type: type, can_be_empty: bool) -> bool:
+    """
+    Checks weather object is list
+    that contains objects of certain type
+    """
+    if not isinstance(user_input, list):
+        return False
+    if not user_input and can_be_empty is False:
+        return False
+    for element in user_input:
+        if not isinstance(element, element_type):
+            return False
+    return True
+
+
+def check_dict (user_input:Any, key_type: type, value_type: type, can_be_empty: bool) -> bool:
+    if not isinstance(user_input, dict):
+        return False
+    if not user_input and can_be_empty is False:
+        return False
+    for key, value in user_input.items():
+        if not (isinstance(key, key_type) and isinstance(value, value_type)):
+            return False
+    return True
+
+
 def extract_phrases(text: str) -> Optional[Sequence[str]]:
```

Comment:
in task description, it says that we should divide a text into phrases by punctuation marks that are located **outside** of the word

does your function handle these cases correctly?